eslint.config.cjs

```diff
@@ -0,0 +1,167 @@
+const typescriptEslint = require('@typescript-eslint/eslint-plugin')
+const jest = require('eslint-plugin-jest')
+const simpleImportSort = require('eslint-plugin-simple-import-sort')
+const _import = require('eslint-plugin-import')
+const localRules = require('eslint-plugin-local-rules')
+const path = require('path')
+
+const { fixupPluginRules } = require('@eslint/compat')
+
+const globals = require('globals')
+const tsParser = require('@typescript-eslint/parser')
+const js = require('@eslint/js')
+
+const { FlatCompat } = require('@eslint/eslintrc')
+
+const compat = new FlatCompat({
+  baseDirectory: __dirname,
+  recommendedConfig: js.configs.recommended,
+  allConfig: js.configs.all,
+})
+
+const project = path.resolve(__dirname, 'tsconfig.json')
+
+module.exports = [
+  {
+    files: ['**/*.ts'],
+  },
+  {
+    ignores: [
+      '.prisma',
+      '.github',
+      'helpers/blaze/**',
+      '.github/renovate.json',
+      '**/dist/**',
+      '**/esm/**',
+      '**/build/**',
+      '**/fixtures/**',
+      '**/byline.ts',
+      '**/prism.ts',
+      '**/charm.ts',
+      '**/pnpm-lock.yaml',
+      '**/generated-dmmf.ts',
+      '**/packages/client/generator-build/**',
+      '**/packages/client/declaration/**',
+      '**/packages/client/runtime/**',
+      '**/packages/client/src/__tests__/types/**',
+      'packages/client/scripts/default-index.js',
+      '**/packages/cli/prisma-client/**',
+      '**/packages/cli/install/**',
+      '**/packages/cli/preinstall/**',
+      'packages/cli/**/tmp-*',
+      '**/sandbox/**',
+    ],
+  },
+  ...compat.extends(
+    'eslint:recommended',
+    'plugin:@typescript-eslint/eslint-recommended',
+    'plugin:@typescript-eslint/recommended',
+    'plugin:@typescript-eslint/recommended-requiring-type-checking',
+    'plugin:prettier/recommended',
+    'plugin:jest/recommended',
+  ),
```

Comment:
I see only one problem in this part. eslintrc will be long not supported and not maintened again. The compat "transforms" the configuration to support eslintrc old configuration. In this case i really want the best thing is to adapt fully to new eslint format.

Reply:
That's a good approach in the long run. For right now, there are still too many plugins that don't officially support the new config format, so we need this. You can always revisit later as plugins are updated.

Reply:
TypeScript eslint now support Eslint v9. Here's a explain https://github.com/typescript-eslint/typescript-eslint/issues/8211#issuecomment-2106427365

Prettier too have a Eslint v9 support. Here's a merged PR https://github.com/prettier/eslint-plugin-prettier/pull/616

And you can see all supported parsers/plugins in the issue [18391](https://github.com/eslint/eslint/issues/18391)